uniffi_bindgen/src/bindings/mod.rs

```diff
@@ -59,8 +61,39 @@ impl TryFrom<String> for TargetLanguage {
     }
 }
 
+#[derive(Debug, Clone, Default, Serialize, Deserialize)]
+pub struct Config {
+    #[serde(default)]
+    kotlin: kotlin::Config,
+    #[serde(default)]
+    swift: swift::Config,
+    #[serde(default)]
+    python: python::Config,
+}
+
+impl From<&ComponentInterface> for Config {
+    fn from(ci: &ComponentInterface) -> Self {
+        Config {
+            kotlin: ci.into(),
+            swift: ci.into(),
+            python: ci.into(),
+        }
+    }
+}
+
+impl MergeWith for Config {
+    fn merge_with(&self, other: &Self) -> Self {
+        Config {
+            kotlin: self.kotlin.merge_with(&other.kotlin),
+            swift: self.swift.merge_with(&other.swift),
+            python: self.python.merge_with(&other.python),
+        }
+    }
+}
+
```

Comment:
This is pretty much the pattern: we define a struct that is `Deserializable`, give an implementation from `ComponentInterface`, and how we're going to merge the two together.